src/goto-symex/symex_goto.cpp

```diff
@@ -345,7 +345,9 @@ void goto_symext::merge_goto(
   statet &state)
 {
   // check atomic section
-  PRECONDITION(state.atomic_section_id == goto_state.atomic_section_id);
+  INVARIANT(
+  	state.atomic_section_id == goto_state.atomic_section_id,
+  	"atomic sections differ across branches");
```

Comment:
The description here is not quite right - the invariant is more a statement that "atomic section should be the same across branches". The description is a description of what invariant we are expecting to hold, not a description of what has gone wrong.